packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -167,6 +179,16 @@ export class HttpApi extends Resource implements IHttpApi {
         httpApi: this,
         autoDeploy: true,
       });
+      if (props?.defaultDomainMapping) {
+        const options = props.defaultDomainMapping;
+        new HttpApiMapping(this, `${options.domainName}${options.mappingKey}`, {
+          api: this,
+          domainName: options.domainName,
+          // use '$default' stage when we create a new API with defaultDomainMapping
+          stage: this.defaultStage!,
+          apiMappingKey: '/',
+        });
+      }
```

Comment:
Instead of creating a new `HttpApiMapping` here, just pass the defaults of defaultDomainMapping to  the `domainMapping` property of `HttpStage` above.